### PR TITLE
[HttpKernel] Deprecate EnvParametersResource

### PR DESCRIPTION
**Pull request by ro0NL**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | yes
| Tests pass?   | yes
| Fixed tickets | #... 
| License       | MIT
| Doc PR        | symfony/symfony-docs#... 

See #24067

### CODE REVIEW

File: src/Symfony/Component/HttpKernel/Kernel.php [hunk 1]
```diff
@@ -13,6 +13,7 @@
 
 use Symfony\Bridge\ProxyManager\LazyProxy\Instantiator\RuntimeInstantiator;
 use Symfony\Bridge\ProxyManager\LazyProxy\PhpDumper\ProxyDumper;
+use Symfony\Component\Config\Resource\EnvParametersResource;
```
Comment (nicolas-grekas) [inline src/Symfony/Component/HttpKernel/Kernel.php]: please revert ordering changes

Comment (nicolas-grekas) [inline src/Symfony/Component/HttpKernel/Kernel.php]: please revert ordering changes

**Comment by ro0NL** [comment]
Do you propose to soft deprecate it in 3.4? Core would still use it.

**Comment by nicolas-grekas** [comment]
Sounds good yes

**Comment by ro0NL** [comment]
Done.

**Review by nicolas-grekas — APPROVED** [review]

**Comment by chalasr** [comment]
Can you add an entry in UPGRADE-4.0 about the removal?

**Comment by ro0NL** [comment]
If this goes first, i can do that in #24067. Along with the actual removal :)

**Comment by chalasr** [comment]
I'd say no, just to not forgot about it.

**Review by chalasr — APPROVED** [review]

**Comment by ro0NL** [comment]
Which raises the question; do we need that file in 3.4 anyway? If you should follow the master version

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
I would update the 4.0 UPGRADE file in this PR, that's what we've been doing for years :)

**Comment by chalasr** [comment]
@ro0NL If the author doesn't do the removing PR (or if we remove while merging up) it's good to have this file already up to date.

**Comment by ro0NL** [comment]
> it's good to have this file already up to date.

Yeah but without the actual code change it's technically not true (yet) :) I see the POV, and as long as lowest branch is leading there's no real issue. Just wanted to point out it can be fragile, and maintaining a single version of truth might make sense :)

@fabpot it already went wrong :) https://www.diffchecker.com/vQeHcX37

basically the same file is out of sync between branches.

Updated anyway :)

**Comment by nicolas-grekas** [comment]
Thank you @ro0NL.

**Pull request closed without merging**